contracts/token/ERC721/ERC721.sol

```diff
@@ -114,12 +114,10 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
      */
     function approve(address to, uint256 tokenId) public virtual {
```

Comment:
>However, we can no longer approve the zero address because of the ownerOf in _approve

I don't think this is true. `_approve(to=address(0), tokenId)` should work normally.

Reply:
You're right, I misinterpreted the semantics of `_approve`. I thought we were restricting `to` instead of `tokenId`'s owner.

Solving.